support/azure/azure-kubernetes/availability-performance/node-auto-repair-errors.md

```diff
@@ -0,0 +1,38 @@
+
+| Error code | Causes & Solution |
+|---|---|
+| ARM ErrorCode: VMExtensionProvisioningError | One or more VM extensions failed to be provisioned on the VM. Read more on possible error types and troubleshooting steps at [Troubleshoot the ERR_VHD_FILE_NOT_FOUND error code (124)](../create-upgrade-delete/error-code-vhdfilenotfound.md). To determine the exact VM extension provisioning error on your node, read more at [View error details in the Azure portal](../create-upgrade-delete/troubleshoot-aks-cluster-creation-issues.md#view-resources-in-the-azure-portal). |
+| ARM ErrorCode: InvalidParameter | This occurs when the node auto-repair process requests a VM that no longer exists. [more guidance] |
+| scaleSetNameAndInstanceIDFromProviderID failed | This is an internal issue where the node may not have been provisioned correctly. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
+| ManagedIdentityCredential authentication failed | This is likely due to an internal issue where your nodes were not initialized properly. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
+| ARM ErrorCode: VMRedeploymentFailed | There was an error with redeploying your node. This may cause your nodepool to enter a Failed state. Read more on potential causes and next steps at [Troubleshoot Azure Kubernetes Service clusters or nodes in a failed state](./cluster-node-virtual-machine-failed-state.md#scenario-3-node-pool-is-in-a-failed-state). |
+| ARM ErrorCode: TooManyVMRedeploymentRequests | This error occurs when your cluster exceeds the internal limit for VM redeployment requests. Redeploy is one of the node auto-repair actions, which means that we are unable to repair your node using redeploy actions. We recommend that you  troubleshoot your node's NotReady status to understand the root cause and resolve the node health issue. Read more at [Basic troubleshooting of Node Not Ready failures](./node-not-ready-basic-troubleshooting.md). |
+| ARM ErrorCode: OutboundConnectivityNotEnabledOnVMSS | This error occurs when your node or overall Virtual Machine Scale Set (VMSS) does not have outbound access enabled. Enable secure outbound access for your VMSS by using a method that's best suited for your application. Read more about the methods at [OutboundConnectivityNotEnabledOnVM. No outbound connectivity configured for virtual machine](../../virtual-machine-scale-sets/deploy/vmss-outbound-connectivity-not-enabled.md#solution). |
+
+## Best practices for monitoring
+- By default, AKS stores Kubernetes events from the past 1 hour. We recommend for you to enable [Container Insights](/azure/azure-monitor/containers/kubernetes-monitoring-enable#enable-container-insights) to store events for up to 90 days. Enabling Container Insights will also allow you to [query events and configure alerts](/azure/aks/events#automating-event-notifications) to quickly detect node auto-repair errors.
```

Comment:
```suggestion
- By default, AKS stores Kubernetes events from the past hour. We recommend that you enable [Container Insights](/azure/azure-monitor/containers/kubernetes-monitoring-enable#enable-container-insights) so that you can store events for up to 90 days. Enabling Container Insights will also allow you to [query events and configure alerts](/azure/aks/events#automating-event-notifications) to quickly detect node auto-repair errors.
```